test/functional/android/find_by_accessibility_id_tests.py

```diff
@@ -14,38 +14,36 @@
 
 import unittest
 
-from appium import webdriver
+from appium.webdriver.common.mobileby import MobileBy
 
-from .helper import desired_capabilities
+from .helper.test_helper import BaseTest, is_ci, wait_for_element
 
 
-class FindByAccessibilityIDTests(unittest.TestCase):
-    def setUp(self):
-        desired_caps = desired_capabilities.get_desired_capabilities('ApiDemos-debug.apk')
-        self.driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)
-
-    def tearDown(self):
-        self.driver.quit()
-
+class FindByAccessibilityIDTests(BaseTest):
     def test_find_single_element(self):
-        self.driver.find_element_by_android_uiautomator('new UiSelector().text("Accessibility")').click()
-        self.driver.find_element_by_android_uiautomator('new UiSelector().text("Accessibility Node Querying")').click()
-        el = self.driver.find_element_by_accessibility_id('Task Take out Trash')
+        wait_for_element(self.driver, MobileBy.ANDROID_UIAUTOMATOR, 'new UiSelector().text("Accessibility")').click()
+        wait_for_element(self.driver, MobileBy.ANDROID_UIAUTOMATOR,
+                         'new UiSelector().text("Accessibility Node Querying")').click()
+        el = wait_for_element(self.driver, MobileBy.ACCESSIBILITY_ID, 'Task Take out Trash')
         self.assertIsNotNone(el)
 
     def test_find_multiple_elements(self):
         els = self.driver.find_elements_by_accessibility_id('Accessibility')
         self.assertIsInstance(els, list)
 
     def test_element_find_single_element(self):
-        self.driver.find_element_by_android_uiautomator('new UiSelector().text("Accessibility")').click()
-        self.driver.find_element_by_android_uiautomator('new UiSelector().text("Accessibility Node Querying")').click()
-        el = self.driver.find_element_by_class_name('android.widget.ListView')
+        if is_ci():
```

Comment:
Did you find the cause?
I guess the uiautomator works fine if the app launched successfully.
Or uiautomator did not work in sometimes..

Reply:
> Did you find the cause?
No for now.
Actually I can say the uiautomator works fine since other cases are ok.
And this case is ok on almost executions, but sometimes failed...

I'll look into more details after this PR.